ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -686,39 +686,33 @@ impl<'a> TraceBuilder<'a> {
         // First apply the constant offset.
         let mut jit_ptr = self.handle_operand(ptr)?;
         if const_off != 0 {
-            let co_ty = jit_ir::IntegerTy::new(32);
-            let co_const =
-                co_ty.make_constant(&mut self.jit_mod, i32::try_from(const_off).unwrap())?;
-            let co_opnd = jit_ir::Operand::Const(self.jit_mod.insert_const(co_const)?);
+            let off_i32 = i32::try_from(const_off).map_err(|_| {
+                CompilationError::LimitExceeded("ptradd constant offset doesn't fit in i32".into())
+            })?;
             jit_ptr = self
                 .jit_mod
-                .push_and_make_operand(jit_ir::PtrAddInst::new(jit_ptr, co_opnd).into())?;
+                .push_and_make_operand(jit_ir::PtrAddInst::new(jit_ptr, off_i32).into())?;
         }
 
         // Now apply any dynamic indices.
         //
         // Each offset is the number of elements multiplied by the byte size of an element.
-        let usize_bitsize = u32::try_from(mem::size_of::<usize>() * 8).unwrap(); // always fits.
-        let count_ty = jit_ir::IntegerTy::new(usize_bitsize);
-        for (count, size) in dyn_elem_counts.iter().zip(dyn_elem_sizes) {
-            let count_opnd = self.handle_operand(count)?;
+        for (num_elems, elem_size) in dyn_elem_counts.iter().zip(dyn_elem_sizes) {
+            let num_elems_opnd = self.handle_operand(num_elems)?;
```

Comment:
No need to change anything in this PR, but one of the nice things about Rust is to overwrite the variable, which stops us worrying about new (longer) variable names. So I'd just do `let num_elems = self.handle_operand(num_elems)?` here (and we do that in various other places).

Reply:
Yeah true, since I'm gonna be squashing later anyway could sneak that in now.

Reply:
I won't object if you do!

Reply:
Fixed in e8233ae45a468ab7797b1a58ffcc09d6951d6ba7